includes/connect/class-wc-stripe-connect.php

```diff
@@ -0,0 +1,185 @@
+				exit;
+			}
+
+		}
+
+		/**
+		 * Saves stripe keys after OAuth response
+		 *
+		 * @param  array $result OAuth response result.
+		 *
+		 * @return array|WP_Error
+		 */
+		private function save_stripe_keys( $result ) {
+
+			if ( ! isset( $result->publishableKey, $result->secretKey ) ) { // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				return new WP_Error( 'Invalid credentials received from WooCommerce Connect server' );
+			}
+
+			$is_test                                = false !== strpos( $result->publishableKey, '_test_' ); // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$prefix                                 = $is_test ? 'test_' : '';
+			$default_options                        = $this->get_default_stripe_config();
+			$options                                = array_merge( $default_options, get_option( self::SETTINGS_OPTION, array() ) );
+			$options['enabled']                     = 'yes';
+			$options['testmode']                    = $is_test ? 'yes' : 'no';
+			$options[ $prefix . 'publishable_key' ] = $result->publishableKey; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+			$options[ $prefix . 'secret_key' ]      = $result->secretKey; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+			// While we are at it, let's also clear the account_id and
+			// test_account_id if present.
+			unset( $options['account_id'] );
+			unset( $options['test_account_id'] );
+
+			update_option( self::SETTINGS_OPTION, $options );
+
+			return $result;
+		}
+
+		/**
+		 * Clears keys for test or production (whichever is presently enabled).
+		 */
+		private function clear_stripe_keys() {
+
+			$options = get_option( self::SETTINGS_OPTION, array() );
+
+			if ( 'yes' === $options['testmode'] ) {
+				$options['test_publishable_key'] = '';
+				$options['test_secret_key']      = '';
+			} else {
+				$options['publishable_key'] = '';
+				$options['secret_key']      = '';
+			}
+
+			// While we are at it, let's also clear the account_id and
+			// test_account_id if present.
+			unset( $options['account_id'] );
+			unset( $options['test_account_id'] );
+
+			update_option( self::SETTINGS_OPTION, $options );
+
+		}
+
+		/**
+		 * Gets default Stripe settings
+		 */
+		private function get_default_stripe_config() {
+
+			$result = array();
+			$gateway = new WC_Gateway_Stripe();
+			foreach ( $gateway->form_fields as $key => $value ) {
+				if ( isset( $value['default'] ) ) {
+					$result[ $key ] = $value['default'];
+				}
+			}
+
+			return $result;
+		}
+
+		public function is_connected() {
+
+			$options = get_option( self::SETTINGS_OPTION, array() );
+
+			if ( isset( $options['testmode'] ) && 'yes' === $options['testmode'] ) {
+				return isset( $options['test_publishable_key'], $options['test_secret_key'] ) && $options['test_publishable_key'] && $options['test_secret_key'];
```

Comment:
This will probably work, but I'd feel more confident if we trimmed before testing for falsiness.

Reply:
I had assumed these values would have any whitespace removed by the time they are set. But further trimming can't hurt! Added in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1267/commits/4c66a0801c9d8bbcb645bb3aec3646eecfd92e25